src/vivarium_csu_hypertension_sdc/components/hypertension_intervention.py

```diff
@@ -0,0 +1,159 @@
+
+        durations = utilities.get_durations_in_range(self.randomness['followup_scheduling'],
+                                           low=0, high=3*28,
+                                           index=sims_on_tx)
+        initialize.loc[sims_on_tx, 'followup_date'] = durations + self.sim_start
+        initialize.loc[sims_on_tx, 'last_visit_date'] = self.sim_start - durations
+
+        self.population_view.update(initialize)
+
+    def on_time_step(self, event):
+        pop = self.population_view.get(event.index)
+
+        followup_scheduled = (self.clock() < pop.followup_date) & (pop.followup_date <= event.time)
+
+        self.attend_followup(pop.index[followup_scheduled], event.time)
+        pop.loc[followup_scheduled, 'last_visit_type'] = 'follow_up'
+
+        background_eligible = pop.index[~followup_scheduled]
+        background_attending = (self.randomness['background_visit_attendance']
+                                .filter_for_rate(background_eligible,
+                                                 self.healthcare_utilization(background_eligible).values))
+
+        self.attend_background(background_attending, event.time)
+        pop.loc[background_attending, 'last_visit_type'] = 'background'
+
+        pop.loc[background_attending.union(pop[followup_scheduled].index), 'last_visit_date'] = event.time
+        self.population_view.update(pop.loc[:, ['last_visit_type', 'last_visit_date']])
+
+    def attend_followup(self, index, visit_date):
+        sbp_measurements = self.measure_sbp(index, visit_date)
+        eligible_for_tx_mask = sbp_measurements >= HYPERTENSIVE_CONTROLLED_THRESHOLD
+
+        treatment_increase_possible = self.check_treatment_increase_possible(index[eligible_for_tx_mask])
+
+        lost_to_ti = self.randomness['therapeutic_inertia'].filter_for_probability(treatment_increase_possible,
+                                                                                   np.tile(self.ti_probability,
+                                                                                           len(treatment_increase_possible)),
+                                                                                   additional_key='lost_to_ti')
+        self.transition_treatment(treatment_increase_possible.difference(lost_to_ti))
+
+        self.schedule_followup(index, visit_date)  # everyone rescheduled no matter whether their tx changed or not
+
+    def attend_background(self, index, visit_date):
+        sbp_measurements = self.measure_sbp(index, visit_date)
+        followup_dates = self.population_view.subview(['followup_date']).get(index).loc[:, 'followup_date']
+
+        eligible_for_tx_mask = (sbp_measurements >= HYPERTENSIVE_CONTROLLED_THRESHOLD) & (followup_dates.isna())
+
+        lost_to_ti = self.randomness['therapeutic_inertia'].filter_for_probability(index[eligible_for_tx_mask],
+                                                                                   np.tile(self.ti_probability,
```

Comment:
Does this not properly broadcast the scalar probability?

Reply:
I don't think so. I was getting an error from where it tries to compare the draws to the probability in randomness that there were mismatched lengths until I did this tiling